src/transformers/api-structure-previews.ts

```diff
@@ -116,13 +150,21 @@ function replaceLinkWithPreview(node: Link | LinkReference) {
     node.children.length > 0 &&
     isTextOrInlineCode(node.children[0])
   ) {
-    hasStructures = true;
-    /* eslint-disable @typescript-eslint/no-explicit-any */
-    (node as any).type = 'jsx';
-    (
-      node as any
-    ).value = `<APIStructurePreview url="${relativeStructurePath}" title="${node.children[0].value}" content="${str}"/>`;
-    /* eslint-enable @typescript-eslint/no-explicit-any */
+    modifiers.add(
+      promise
+        .then((content) => {
+          const previewNode = node as unknown as Literal<string>;
+          previewNode.type = 'jsx';
+          previewNode.value = `<APIStructurePreview url="${relativeStructureUrl}" title="${
+            (node.children[0] as Text | InlineCode).value
+          }" content="${encodeURIComponent(content)}"/>`;
+        })
+        .catch((err) => {
+          logger.error(err);
+          // NOTE - if build starts failing, comment the throw out
+          throw err;
```

Comment:
Opting to break loudly and break the build for the moment, otherwise we'll never notice an error being logged. If this becomes a problem we can re-evaluate.